### PR TITLE
Add velocityThres param for torque control

### PR DESCRIPTION
**Pull request by sgiraz**
**What's new:**
- Add a new `velocityThres` parameter for friction compensation used during torque control. This parameters is used to enable the new torque control law and fine-tune the friction compensation at FW level.
- Remove unused `bemf` and `bemf_scale` parameter from `MotorTorqueParameters`

> **Note**
> - Tested on a single joint setup
> - Not Tested on a real robot yet

### CODE REVIEW

File: src/devices/controlBoard_nws_yarp/RPCMessagesParser.cpp [hunk 1]
```diff
@@ -482,14 +482,13 @@ void RPCMessagesParser::handleTorqueMsg(const yarp::os::Bottle& cmd,
                 break;
             }
 
-            params.bemf = b->get(0).asFloat64();
-            params.bemf_scale = b->get(1).asFloat64();
             params.ktau = b->get(2).asFloat64();
```
Comment (traversaro) [inline src/devices/controlBoard_nws_yarp/RPCMessagesParser.cpp:485]: If we do not add anymore `bemf` and `bemf_scale`, then `ktau` will be the first element of the vector, not the third one. Perhaps we can do the removal of bemf in a separate PR, as it could impact also non-iCub users of YARP?

**Comment by traversaro** [comment]
See the comment in https://github.com/robotology/yarp/pull/3008/files#r1308588443 .

**Comment by pattacini** [comment]
Hi @sgiraz 

There may be benefits in targeting https://github.com/robotology/yarp/tree/yarp-3.8 rather than `master`. Don't know @randaz81 @traversaro @Nicogene opinions though.

**Comment by randaz81** [comment]
Removal of any API/parameters should be addressed in a separate PR. This PR cannot be approved.

**Review by randaz81 — CHANGES_REQUESTED** [review]
Not compliant with YARP API deprecation guidelines.

**Comment by randaz81** [comment]
> Hi @sgiraz
> 
> There may be benefits in targeting https://github.com/robotology/yarp/tree/yarp-3.8 rather than `master`. Don't know @randaz81 @traversaro @Nicogene opinions though.

yarp 3.8 is bug-fix only, so the correct branch for introducing a new feature is master.

**Comment by randaz81** [comment]
Could you better document in the .h file what velocityThres is (or maybe choose a more appropriate name for the variable?) 
The doxygen comment "velocity threshold for torque control" sounds ambiguous.

**Comment by pattacini** [comment]
> yarp 3.8 is bug-fix only, so the correct branch for introducing a new feature is master.

Right 👍🏻 

>Not compliant with YARP API deprecation guidelines.

Can you point @sgiraz to the guidelines in order for him to follow the correct pathway?
Is maybe the same [suggestion](https://github.com/robotology/yarp/pull/3008#discussion_r1308588443) @traversaro jotted down as a comment?

**Comment by randaz81** [comment]
> Can you point @sgiraz to the guidelines in order for him to follow the correct pathway? Is maybe the same [suggestion](https://github.com/robotology/yarp/pull/3008#discussion_r1308588443) @traversaro jotted down as a comment?

Yes, indeed it's the same comment. The guidelines can be found https://yarp.it/latest/contributing.html. 
Here is a quick recap. Basically, we need:  
- a PR on master to introduce the new parameters/methods.
- a PR on master to mark as deprecated (do not remove) the unused parameters.
- In a future major release we can remove the the deprecated parameters (another PR on master)
